src/Testcontainers/Builders/IAbstractBuilder`3.cs

```diff
@@ -79,6 +79,9 @@ public interface IAbstractBuilder<out TBuilderEntity, out TResourceEntity, out T
     /// <summary>
     /// Allows low level modifications of the Docker.DotNet entity after the builder configuration has been applied. Multiple low level modifications will be executed in order of insertion.
     /// </summary>
+    /// <remarks>
+    /// This API is intended for special use cases only. We do not recommend using it and cannot maintain backwards compatibility.
```

Comment:
@mdelapenya @eddumelendez I've added a note that the parameter modifier is only for special use cases.